monitoring_scripts/featuredrift.py

```diff
@@ -0,0 +1,109 @@
+        Output: DataFrame containing columns 'reference_project_id', 'current_project_id',
+                'reference_dataset_len', 'current_dataset_len', 'drift_share', 'number_of_columns',
+                'number_of_drifted_columns', 'share_of_drifted_columns', 'dataset_drift'
+        """
+        final_result = list()
+        reference_project_ids = list(self.reference_df["project_id"].unique())
+        current_project_ids = list(self.current_df["project_id"].unique())
+
+        for project_id in tqdm(reference_project_ids):
+            if project_id in current_project_ids:
+                reference_df = self._project_id_based_mask(self.reference_df, project_id)
+                current_df = self._project_id_based_mask(self.current_df, project_id)
+                
+                reference_embedding_lst = self._process_embeddings(reference_df["embeddings"])
+                current_embedding_lst = self._process_embeddings(current_df["embeddings"])
+
+                reference_df = pd.DataFrame(reference_embedding_lst).sample(n=n_samples, random_state=random_state)
+                current_df = pd.DataFrame(current_embedding_lst).sample(n=n_samples, random_state=random_state)
+
+                if len(reference_df) and len(current_df):
+                    self.data_drift_dataset_report.run(
+                        reference_data=reference_df,
+                        current_data=current_df
+                    )
+
+                    temp_result = {}
+                    temp_result["reference_project_id"] = project_id
+                    temp_result["current_project_id"] = project_id
+                    temp_result["reference_dataset_len"] = len(reference_df)
+                    temp_result["current_dataset_len"] = len(current_df)
+                    data_drift_report = self.data_drift_dataset_report.as_dict()
+                    # Note the keys might change in the future
+                    temp_result["drift_share"] = data_drift_report["metrics"][0]["result"]["drift_share"]
+                    temp_result["number_of_columns"] = data_drift_report["metrics"][0]["result"]["number_of_columns"]
+                    temp_result["number_of_drifted_columns"] = data_drift_report["metrics"][0]["result"]["number_of_drifted_columns"]
+                    temp_result["share_of_drifted_columns"] = temp_result["drift_share"] = data_drift_report["metrics"][0]["result"]["share_of_drifted_columns"]
+                    temp_result["dataset_drift"] = temp_result["drift_share"] = data_drift_report["metrics"][0]["result"]["dataset_drift"]
+                    temp_result["generated_at"] = datetime.date.today()
+
+                    final_result.append(temp_result)
+        
+        return pd.DataFrame.from_records(final_result)
+    
+
+if __name__ == "__main__":
+    reference_data_path = "csvfiles/data_with_embeddings_22000.csv"
+    current_data_path = "csvfiles/sampled_data_with_embeddings_testset.csv"
```

Comment:
This file path should be passed differently

Reply:
just used for testing.